packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { createContext, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+export interface AccessTokenContextValue {
+  /**
+   * The returned accessToken wheter from the props or from the fetchToken function
+   */
+  accessToken?: string
+  /**
+   * If true, the access token is being fetched
+   */
+  isLoading?: boolean
+  /**
+   * Function that will be called when the access token expires.
+   */
+  onExpiredToken?: () => void
+  /**
+   * If true, the access token failed to be fetched after the maximum retries
+   */
+  failedRetry?: boolean
```

Comment:
How about making this an optional `error` property (like react-query) which contains the last error we encountered when fetching the token? If we successfully fetch the token, clear it out.
```suggestion
  /**
   * If present, the AccessTokenProvider encountered an error fetching the access token. This will be set to undefined as soon as fetching the access token succeeds.
   */
  error?: Error
```